src/Spryker/Zed/Url/Business/Url/UrlCreator.php

```diff
@@ -115,7 +115,9 @@ protected function persistUrlEntity(UrlTransfer $urlTransfer)
         $urlTransfer->requireUrl();
         $urlEntity = $this->urlQueryContainer->queryUrl($urlTransfer->getUrl())->findOneOrCreate();
 
+        $id = $urlEntity->getIdUrl();
```

Comment:
with this, you're merging `read from db` and `write to DB` code.
please make sure to read dat from DB first, then modify, and only then call this method.

Reply:
I only save temporarily the id created by findOneOrCreate and reset it after it was removed with null from transfer. The mix with read from db and write to db is already present in the untouched function. And the ID is needed since it is required in /**
 * @param \Generated\Shared\Transfer\UrlTransfer $urlTransfer
 *
 * @return \Generated\Shared\Transfer\UrlTransfer
 */
protected function executeCreateUrlTransaction(UrlTransfer $urlTransfer): UrlTransfer
{
    $this->notifyBeforeSaveObservers($urlTransfer);
    $urlTransfer = $this->persistUrlEntity($urlTransfer);
    **$this->urlActivator->activateUrl($urlTransfer);**
    $this->notifyAfterSaveObservers($urlTransfer);
    return $urlTransfer;
}

Reply:
@julianzimmermann I have investigated the issue and see that the transfer this method returns always has the ID. Even if you manually set null as ID to the transfer before calling this method, it will be rewritten by this line `$urlTransfer->fromArray($urlEntity->toArray(), true);` since the ID will be restored after calling the `save()` method. Could you please provide additional information about the issue you faced so we can found the root of the issue and fix it?

Reply:
Looks like I have caught the issue. It appears in case if you manually set `$urlTransfer->setIdUrl(null);` and the URL already exists in the DB. But I think instead of updating the method to save the ID in a temporary variable, you should avoid setting the ID as null before calling the method because in this case, any important field might be lost if you set it to the transfer before, not only the ID.

Reply:
Thank you for investigation. We had some issues with the URLs in our multi store concept. For example, we could not save same URLs for different stores, URLs stayed in the db and redis after remove product from store and was not able to readd again etc. So we had to do some customize work to get it working for our case. So in our case the $urlTransfer comes with idUrl = null. Anyway, we fixed it for us in our module but maybe others have same issues.